### PR TITLE
Entities, DTOs and converters creation

### PR DESCRIPTION
**Pull request by AngheloZambrana**
# Pull Request Description

This pull request introduces the implementation of converters for the entities in your project, as well as the creation of corresponding DTOs and entities. The following changes have been made:

1. **Converter Interfaces**:
   - `IConverter1To1`: Conversion from entity to DTO (1 to 1).
   - `IConverter1To2`: Conversion from entity to DTO using two inputs (2 to 1).
   - `IConverter2To1`: Conversion from two inputs to a single DTO (2 to 1).
   - `IConverter2To2`: Conversion from two inputs to two DTOs (2 to 2).
   - `IConverter1To3`: Conversion from entity to DTO using three inputs (1 to 3).
   - `IConverter1To4`: Conversion from entity to DTO using four inputs (1 to 4).

2. **Implemented Converters**:
   - `ContactConverter`
   - `HotelConverter` (implements `IConverter1To4`)
   - `ReservationConverter`
   - `RoomConverter`
   - `UserConverter`

3. **DTOs and Entities**:
   - Corresponding DTOs and entities have been created and updated to reflect the relationships between them.

Reviewers: @JCarlosHidalgoSosa @OropezaHugo

### CODE REVIEW

**Review by JCarlosHidalgoSosa — APPROVED** [review]

File: backend/Converters/ContactConverter.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using DTO;
+using DefaultNamespace;
```
Comment (OropezaHugo) [inline backend/Converters/ContactConverter.cs]: fix DefaultNamespaces in the PR to its folder's namespace

File: backend/Converters/ContactConverter.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            return new ContactDTO
+            {
+                ContactID = contact.ContactID,
+                PhoneNumber = contact.PhoneNumber,
+                Email = contact.Email,
+                ReservationList = reservationList
+            };
+        }
+    }
+}
```
Comment (OropezaHugo) [inline backend/Converters/ContactConverter.cs]: add endline to the files

File: backend/Converters/ContactConverter.cs (hunk 1 above)
Comment (AngheloZambrana) [reply backend/Converters/ContactConverter.cs]: I already change that, can you review again?

**Review by OropezaHugo — APPROVED** [review]

File: backend/DTOs/ReservationDTO.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+namespace DTOs;
+
+public class ReservationDTO
+{
+    // Reservation properties
+    public Guid UserID { get; set; }
```
Comment (OropezaHugo) [inline backend/DTOs/ReservationDTO.cs]: please update UserId to the ContactID as it is set in the DB diagram

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**